runtime/auth/aws-credentials/common/src/aws/smithy/kotlin/runtime/auth/awscredentials/CredentialsProvider.kt

```diff
@@ -4,12 +4,18 @@
  */
 package aws.smithy.kotlin.runtime.auth.awscredentials
 
+import aws.smithy.kotlin.runtime.tracing.NoOpTraceSpan
+import aws.smithy.kotlin.runtime.tracing.TraceSpan
+
 /**
  * Represents a producer/source of AWS credentials
  */
 public interface CredentialsProvider {
     /**
      * Request credentials from the provider
+     * @param traceSpan An optional span to which trace events should be emitted. Defaults to [NoOpTraceSpan] if none is
+     * specified.
+     * @return A set of [Credentials].
      */
-    public suspend fun getCredentials(): Credentials
+    public suspend fun getCredentials(traceSpan: TraceSpan = NoOpTraceSpan): Credentials
```

Comment:
Follow-up items:
* Investigate places where non-suspend logging is happening, how would we wire through `CoroutineContext` if we decided
* Investigate wiring `ExecutionContext` through instead of `TraceSpan` directly